tests/stree_optimizer/test_optimization.py

```diff
@@ -56,14 +63,63 @@ def test_stree_roundtrip_no_opt() -> None:
     in_qty = quantity_factory.ones([X_DIM, Y_DIM, Z_DIM], "")
     out_qty = quantity_factory.zeros([X_DIM, Y_DIM, Z_DIM], "")
 
-    # Temporarily flip the internal switch
-    import ndsl.dsl.dace.orchestration as orch
+    with StreeOptimization():
+        code(in_qty, out_qty)
 
-    orch._INTERNAL__SCHEDULE_TREE_OPTIMIZATION = True
-    orch._INTERNAL__SCHEDULE_TREE_PASSES = [CartesianAxisMerge(AxisIterator._K)]
+        assert (
+            len(stencil_factory.config.dace_config.loaded_precompiled_SDFG.values())
+            == 1
+        )
+        sdfg = list(
+            stencil_factory.config.dace_config.loaded_precompiled_SDFG.values()
+        )[0]
+        all_maps = [
+            (me, state)
+            for me, state in sdfg.sdfg.all_nodes_recursive()
+            if isinstance(me, dace.nodes.MapEntry)
+        ]
+
+        assert len(all_maps) == 3
+        assert (out_qty.field[:] == 4).all()
+
+
+class LocalRefineableCode(NDSLRuntime):
+    def __init__(
+        self, stencil_factory: StencilFactory, quantity_factory: QuantityFactory
+    ) -> None:
+        super().__init__(stencil_factory.config.dace_config)
+        self.stencil_A = stencil_factory.from_dims_halo(
+            func=stencil_A,
+            compute_dims=[X_DIM, Y_DIM, Z_DIM],
+        )
+        self.tmp = self.make_local(quantity_factory, [X_DIM, Y_DIM, Z_DIM])
+
+    def __call__(self, in_field: Quantity, out_field: Quantity) -> None:
+        self.stencil_A(in_field, self.tmp)
+        self.stencil_A(self.tmp, out_field)
```

Comment:
The fact that this works is so cool! We open up a world of options with true cross-stencil optimization at that level